```
C:\__w\1\s\common\include\pcl\impl\point_types.hpp(539,35): error C1001: Internal compiler error. [C:\build\common\pcl_common.vcxproj]
C:\__w\1\s\common\include\pcl\impl\point_types.hpp(539,35): error C1001: (compiler file 'D:\a\_work\1\s\src\vctools\Compiler\CxxFE\sl\p1\c\p1_init.c', line 13000) [C:\build\common\pcl_common.vcxproj]
C:\__w\1\s\common\include\pcl\impl\point_types.hpp(539,35): error C1001:  To work around this problem, try simplifying or changing the program near the locations listed above. [C:\build\common\pcl_common.vcxproj]
C:\__w\1\s\common\include\pcl\impl\point_types.hpp(539,35): error C1001: If possible please provide a repro here: https://developercommunity.visualstudio.com/  [C:\build\common\pcl_common.vcxproj]
C:\__w\1\s\common\include\pcl\impl\point_types.hpp(539,35): error C1001: Please choose the Technical Support command on the Visual C++  [C:\build\common\pcl_common.vcxproj]
C:\__w\1\s\common\include\pcl\impl\point_types.hpp(539,35): error C1001:  Help menu, or open the Technical Support help file for more information [C:\build\common\pcl_common.vcxproj]
  (compiling source file '../../__w/1/s/common/src/point_types.cpp')
```

wat

--- a/common/include/pcl/impl/point_types.hpp
+++ b/common/include/pcl/impl/point_types.hpp
@@ -210,13 +210,12 @@ namespace pcl
 
 #define PCL_ADD_UNION_POINT4D \
   union EIGEN_ALIGN16 { \
-    float data[4]; \
+      /* ensure last homogeneous component is 1 for affine transformations */ \
+    float data[4] {0, 0, 0, 1}; \
     struct { \
       float x; \
       float y; \
       float z; \
-      /* ensure last homogeneous component is 1 for affine transformations */ \
-      float _w  = 1; \
     }; \
   };
 

